gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManager.java

```diff
@@ -0,0 +1,430 @@
+    this.jobStatusRetriever = jobStatusRetriever;
+    TimeUnit timeUnit = TimeUnit.valueOf(ConfigUtils.getString(config, FAILED_DAG_RETENTION_TIME_UNIT, DEFAULT_FAILED_DAG_RETENTION_TIME_UNIT));
+    this.failedDagRetentionTime = timeUnit.toMillis(ConfigUtils.getLong(config, FAILED_DAG_RETENTION_TIME, DEFAULT_FAILED_DAG_RETENTION_TIME));
+  }
+
+  public void setActive(boolean active) throws IOException {
+    if (this.isActive == active) {
+      log.info("DagManager already {}, skipping further actions.", (!active) ? "inactive" : "active");
+    }
+    this.isActive = active;
+    try {
+      if (this.isActive) {
+        log.info("Activating NewDagManager.");
+        //Initializing state store for persisting Dags.
+        this.dagStateStore = createDagStateStore(config, topologySpecMap);
+        // todo - implement as a kill dag action
+        this.failedDagStateStore = createDagStateStore(ConfigUtils.getConfigOrEmpty(config, FAILED_DAG_STATESTORE_PREFIX).withFallback(config),
+            topologySpecMap);
+        this.failedDagIds = Collections.synchronizedSet(failedDagStateStore.getDagIds());
+        dagManagerMetrics.activate();
+        userQuotaManager.init(dagStateStore.getDags());
+        DagManager.FailedDagRetentionThread failedDagRetentionThread =
+            new DagManager.FailedDagRetentionThread(failedDagStateStore, failedDagIds, failedDagRetentionTime);
+        this.scheduledExecutorPool.scheduleAtFixedRate(failedDagRetentionThread, 0, retentionPollingInterval, TimeUnit.MINUTES);
+        loadDagFromDagStateStore();
+        this.houseKeepingThreadPool = Executors.newSingleThreadScheduledExecutor(new NamedThreadFactory("LoadDagsThread"));
+        for (int delay = INITIAL_HOUSEKEEPING_THREAD_DELAY; delay < MAX_HOUSEKEEPING_THREAD_DELAY; delay *= 2) {
+          this.houseKeepingThreadPool.schedule(() -> {
+            try {
+              loadDagFromDagStateStore();
+            } catch (Exception e) {
+              log.error("failed to sync dag state store due to ", e);
+            }
+          }, delay, TimeUnit.MINUTES);
+        }
+      } else { //Mark the DagManager inactive.
+        log.info("Inactivating the DagManager. Shutting down all DagManager threads");
+        this.scheduledExecutorPool.shutdown();
+        dagManagerMetrics.cleanup();
+        this.houseKeepingThreadPool.shutdown();
+        try {
+          this.scheduledExecutorPool.awaitTermination(TERMINATION_TIMEOUT, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+          log.error("Exception encountered when shutting down DagManager threads.", e);
+        }
+      }
+    } catch (IOException e) {
+        log.error("Exception encountered when activating the new DagManager", e);
+        throw new RuntimeException(e);
+    }
+  }
+
+  public static DagStateStore createDagStateStore(Config config, Map<URI, TopologySpec> topologySpecMap) {
```

Comment:
my first instinct is that this belongs within `DagManagementStateStore`

...along w/ the various CRUD ops that follow